core/src/main/java/hudson/Util.java

```diff
@@ -1708,8 +1708,24 @@ public static long daysElapsedSince(@NonNull Date date){
     @Restricted(value = NoExternalUse.class)
     static boolean GC_AFTER_FAILED_DELETE = SystemProperties.getBoolean(Util.class.getName() + ".performGCOnFailedDelete");
 
+    /**
+     * The maximum number of errors that are permitted during a single attempt
+     * to delete a directory before that attempt is aborted.
+     * <p>
+     * Even if one deletion attempt aborts early, additional attempts may occur
+     * depending on the value of {@link DELETION_RETRIES}.
+     * <p>
+     * The purpose of this setting is to prevent the composite exceptions
+     * thrown by failed directory deletion operations from becoming extremely
+     * large. For example, in cases where Jenkins attempts to delete a large
+     * directory that it does not have permission to delete, we do not want the
+     * composite exception to contain an exception for every single file inside
+     * of that directory.
+     */
+    private static final int DELETION_MAX_ERRORS_BEFORE_ABORT = 10;
```

Comment:
I am not sure whether we need to make this configurable by system property, so I did not make it configurable, but I am happy to make it configurable if there is a reasonable case for doing so. I am also happy to increase/decrease the default value, maybe 15 or 25 would be a reasonable default instead.